extensions/wikia/GlobalWatchlist/GlobalWatchlistHooks.class.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+use \Wikia\Tasks\AsyncTaskList;
+
+class GlobalWatchlistHooks {
+
+	public static function getPreferences( $user, &$defaultPreferences ) {
+
+		$defaultPreferences['watchlistdigest'] = [
+			'type' => 'toggle',
+			'label-message' => 'tog-watchlistdigest',
+			'section' => 'watchlist/advancedwatchlist',
+		];
+
+		$defaultPreferences['watchlistdigestclear'] = [
+			'type' => 'toggle',
+			'label-message' => 'tog-watchlistdigestclear',
+			'section' => 'watchlist/advancedwatchlist',
+		];
+		
+		return true;
+	}
+
+	/**
+	 * Check if the user is unsubscribing from either the weekly digest or email altogether and, if so,
+	 * clears all their entries from the global_watchlist table.
+	 * @param $formData
+	 * @param $error
+	 * @return bool
+	 */
+	static public function savePreferences( &$formData, &$error ) {
```

Comment:
Usually access specifiers come first. Please also rearrange this for the other methods.